test/base/Group.cpp

```diff
@@ -74,6 +74,35 @@ BOOST_FIXTURE_TEST_CASE(unionTest, GroupTestFixture)
     BOOST_CHECK(*it == d);
 }
 
+BOOST_FIXTURE_TEST_CASE(multipleGroupsUnionTest, GroupTestFixture)
+{
+    Group<Foo> groups[5];
+    groups[0].Add(c);
+    groups[1].Add(e);
+    groups[2].Add(e);
+    groups[2].Add(d);
+    groups[3].Add(a);
+    groups[3].Add(e);
+    groups[4].Add(b);
+
+    auto unionGroup = Unite(groups[0], groups[1], groups[2], groups[3], groups[4]);
```

Comment:
Why not just
~~~cpp
Group<Foo> groupThree;
groupThree.Add(c);
groupThree.Add(a);

auto unionGroup = Unite(groupOne, groupTwo, groupThree)
~~~
instead of the C-Style array?